ui/app/components/namespace-picker.js

```diff
@@ -64,6 +69,38 @@ export default class NamespacePicker extends Component {
     ];
   }
 
+  #getNamespaceLink(location, namespace) {
+    const origin = this.#getOrigin(location);
+    const encodedNamespace = encodeURIComponent(namespace.path);
+
+    let queryParams = '';
+    if (namespace.path !== '') {
```

Comment:
This may not be really necessary as you convert to TS and have stronger typing, but you could check 

```suggestion
  if (!isEmpty(namespace.path)) {
```
to guard against `undefined` or `null` as well.